### PR TITLE
feat: disable mTLS plans for TCP APIs

### PR DESCRIPTION
**Pull request by dev-marek**
## Issue

https://gravitee.atlassian.net/browse/APIM-6963

## Description

image

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-ckxwdnnwqk.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/plan/use_case/CreatePlanUseCaseTest.java [hunk 1]
```diff
@@ -175,6 +175,20 @@ void should_throw_when_create_a_federated_plan() {
         Assertions.assertThat(throwable).isInstanceOf(PlanInvalidException.class).hasMessage("Can't manually create Federated Plan");
     }
 
+    @Test
+    void should_throw_when_create_mtls_plan_for_tcp_api() {
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/plan/use_case/CreatePlanUseCaseTest.java]:
```suggestion
    void should_throw_when_creating_mtls_plan_for_tcp_api() {
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/plan/use_case/CreatePlanUseCase.java [hunk 2]
```diff
@@ -41,6 +41,10 @@ public Output execute(Input input) {
             throw new PlanInvalidException("Can't manually create Federated Plan");
         }
 
+        if (api.getDefinitionVersion() == DefinitionVersion.V4 && api.getApiDefinitionV4().isTcpProxy()) {
+            throw new PlanInvalidException("Cannot create mTLS plan for TCP API");
+        }
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/plan/use_case/CreatePlanUseCase.java]: I think either your test or error message is wrong here
You are just checking the version of the API and if it is a TCP proxy, but there is no test on the plan type.
So why the message is talking about mTLS plan ?

**Review by ytvnr — CHANGES_REQUESTED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/plan/use_case/CreatePlanUseCase.java (hunk 2 above)
Comment (dev-marek) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/plan/use_case/CreatePlanUseCase.java]: Ok, fixed

**Review by ytvnr — APPROVED** [review]
lgtm

**Pull request merged**